pkg/segment/writer/agiletree.go

```diff
@@ -230,12 +230,16 @@ func (stb *StarTreeBuilder) DropColumns(colsToDrop []string) error {
 	}
 
 	// check if all columns to drop are present
-	for _, colToDrop := range colsToDrop {
-		colIdx, exists := mapColNameToIdx[colToDrop]
-		if !exists {
-			return fmt.Errorf("DropColumns: column to drop %v not found", colToDrop)
+	for colcolsToDrop := range colsToDrop {
+		if _, exists := mapColNameToIdx[colcolsToDrop]; !exists {
+			return fmt.Errorf("DropColumns: column to drop %v not found", colcolsToDrop)
```

Comment:
nit: rename `colcolsToDrop` to `colToDrop`, or `col`, or similar

Reply:
addressed in the latest commit.